addons/mail/models/ir_http.py

```diff
@@ -13,6 +13,8 @@ def session_info(self):
         """Override to add the current user data (partner or guest) if applicable."""
         result = super().session_info()
         store = StoreData()
+        cids = request.httprequest.cookies.get('cids', str(request.env.user.company_id.id))
+        store.add({"Store": {'cids': [int(cid) for cid in cids.split(',')]}})
```

Comment:
Store is intended for values in JS, not for transferring stuff in python.

You should probably update the context on `self` with allowed companies from the cookie in parent before calling `session_info`.